platform/darwin/src/MGLNetworkConfiguration.h

```diff
@@ -1,18 +1,37 @@
 #import <Foundation/Foundation.h>
 
+#import "MGLFoundation.h"
+
 NS_ASSUME_NONNULL_BEGIN
 
 /**
- The MGLNetworkConfiguration object provides a global way to set a base API URL for
- retrieval of map data, styles, and other resources.
-
- Currently, MGLNetworkConfiguration is a private API.
+ The `MGLNetworkConfiguration` object provides a global way to set a base
+ `NSURLSessionConfiguration`, and other resources.
  */
+MGL_EXPORT
 @interface MGLNetworkConfiguration : NSObject
 
-/// Returns the shared instance of the `MGLNetworkConfiguration` class.
+/**
+ Returns the shared instance of the `MGLNetworkConfiguration` class.
+ */
 @property (class, nonatomic, readonly) MGLNetworkConfiguration *sharedManager;
 
+/**
+ The session configuration object that is used by the `NSURLSession` objects
+ in this SDK.
+ 
+ If this property is set to nil or if no session configuration is provided this property
+ is set to the default session configuration.
+ 
+ Assign this object before instantiating any `MGLMapView` object.
+ 
+ @note: `NSURLSession` objects store a copy of this configuration. Any further changes
+ to mutable properties on this configuration object passed to a session’s initializer
+ will not affect the behavior of that session.
+ 
+ */
+@property (strong, null_resettable) NSURLSessionConfiguration *sessionConfiguration;
```

Comment:
Since you're using `@synchronized` below, what do you think to marking this as `atomic`? (essentially documentation)

Reply:
Oh, I am. `@atomic` is the default property if `@nonatomic` is not present. I can add it to make it clear tho.